### PR TITLE
mmteb | Arabic Retrieval Task | SadeemQuestionRetrieval

### PR DESCRIPTION
**Pull request by bakrianoo**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- This is a dataset for `mmteb` initiative. 
- The Dataset is for Arabic Retrieval tasks
- The Dataset is for Question-Based searching tasks (The retrieval part in the RAG pipeline)

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

<details>
<summary> (sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2) Results</summary>

#### Sub intro

```json
{
  "dataset_revision": "3cb0752b182e5d5d740df547748b06663c8e0bd9",
  "mteb_dataset_name": "SadeemQuestionRetrieval",
  "mteb_version": "1.8.2",
  "test": {
    "evaluation_time": 32.78,
    "map_at_1": 0.17329,
    "map_at_10": 0.36268,
    "map_at_100": 0.36891,
    "map_at_1000": 0.36931,
    "map_at_20": 0.36621,
    "map_at_3": 0.34027,
    "map_at_5": 0.35499,
    "mrr_at_1": 0.16419,
    "mrr_at_10": 0.35615,
    "mrr_at_100": 0.3621,
    "mrr_at_1000": 0.3625,
    "mrr_at_20": 0.35937,
    "mrr_at_3": 0.33365,
    "mrr_at_5": 0.34797,
    "ndcg_at_1": 0.17329,
    "ndcg_at_10": 0.44446,
    "ndcg_at_100": 0.47729,
    "ndcg_at_1000": 0.48885,
    "ndcg_at_20": 0.45737,
    "ndcg_at_3": 0.39977,
    "ndcg_at_5": 0.42604,
    "precision_at_1": 0.17329,
    "precision_at_10": 0.06932,
    "precision_at_100": 0.00853,
    "precision_at_1000": 0.00095,
    "precision_at_20": 0.03722,
    "precision_at_3": 0.19116,
    "precision_at_5": 0.12733,
    "recall_at_1": 0.17329,
    "recall_at_10": 0.69315,
    "recall_at_100": 0.85256,
    "recall_at_1000": 0.94543,
    "recall_at_20": 0.74438,
    "recall_at_3": 0.57348,
    "recall_at_5": 0.63667
  }
}
```

</details>

<details>
<summary> (intfloat/multilingual-e5-small) Results</summary>

#### Sub intro

```json
{
  "dataset_revision": "3cb0752b182e5d5d740df547748b06663c8e0bd9",
  "mteb_dataset_name": "SadeemQuestionRetrieval",
  "mteb_version": "1.8.2",
  "test": {
    "evaluation_time": 70.17,
    "map_at_1": 0.32408,
    "map_at_10": 0.58825,
    "map_at_100": 0.58921,
    "map_at_1000": 0.58925,
    "map_at_20": 0.58904,
    "map_at_3": 0.5809,
    "map_at_5": 0.58657,
    "mrr_at_1": 0.30302,
    "mrr_at_10": 0.57539,
    "mrr_at_100": 0.57635,
    "mrr_at_1000": 0.57638,
    "mrr_at_20": 0.57618,
    "mrr_at_3": 0.56909,
    "mrr_at_5": 0.57364,
    "ndcg_at_1": 0.32408,
    "ndcg_at_10": 0.68539,
    "ndcg_at_100": 0.68952,
    "ndcg_at_1000": 0.69052,
    "ndcg_at_20": 0.68821,
    "ndcg_at_3": 0.67141,
    "ndcg_at_5": 0.68144,
    "precision_at_1": 0.32408,
    "precision_at_10": 0.09713,
    "precision_at_100": 0.00989,
    "precision_at_1000": 0.001,
    "precision_at_20": 0.04911,
    "precision_at_3": 0.31179,
    "precision_at_5": 0.19186,
    "recall_at_1": 0.32408,
    "recall_at_10": 0.97128,
    "recall_at_100": 0.98947,
    "recall_at_1000": 0.99761,
    "recall_at_20": 0.98229,
    "recall_at_3": 0.93538,
    "recall_at_5": 0.95931
  }
}
```

</details>

### CODE REVIEW

**Review by isaac-chung — COMMENTED** [review]
Looks good at first glance. Could you commit the results as well please?

**Comment by bakrianoo** [comment]
Hi @isaac-chung 
I attached the results in the PR description above. Can you tell me, where can I commit them to the project?

**Comment by isaac-chung** [comment]
After you run `mteb run -m {model_name} -t {task_name}`, the json file will be automatically added to the `results` folder. You can then commit those.

**Comment by bakrianoo** [comment]
@isaac-chung 
Done. Please check.

**Review by isaac-chung — APPROVED** [review]
Looks good, thanks! Feel free to add points if your want.

File: docs/mmteb/points/643.jsonl [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "bakrianoo", "New dataset": 6, "Running Models": 1}
```
Comment (isaac-chung) [inline docs/mmteb/points/643.jsonl:1]: {"GitHub": "bakrianoo", "New dataset": 2}

Comment (isaac-chung) [inline docs/mmteb/points/643.jsonl:1]:
```suggestion
{"GitHub": "bakrianoo", "New dataset": 2}
```

[the remark this answers is not in the document]
Comment (bakrianoo) [reply docs/mmteb/points/643.jsonl]: Done

**Comment by isaac-chung** [comment]
Hmm I can't help merge main, I think you need to enable maintains to make changes to the PR.

**Comment by bakrianoo** [comment]
@isaac-chung 
Is enabling these options enough?

image

**Comment by isaac-chung** [comment]
It's more related to: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by bakrianoo** [comment]
@isaac-chung 
Since the repo is under organization, not my account, I can not find this option.

I found an open issue here about this
https://github.com/orgs/community/discussions/5634

Do you have any idea how to solve it?

**Comment by isaac-chung** [comment]
Can you see this check box in red? If so, please enable it.

image

**Comment by bakrianoo** [comment]
@isaac-chung 
That's the problem. Since the repo belongs to an organization, this option is not available. 
The only solution I can find now is to cancel this request and do a new one from a personal account.

**Comment by isaac-chung** [comment]
Ah thanks, I see now. And I noticed that the source branch is main, instead of a separate feature branch as well. 

The check box is only for my convenience, as the only things missing in the PR for merging are 1) to run linting and 2) update branch from origin main. 

If you can do those fairly soon then there's no need to close this PR and restart.

**Comment by bakrianoo** [comment]
@isaac-chung 
I synced the main branch and applied `make lint`.
Please advise for any missed steps. 

Sorry for any disturbing :)

**Pull request merged**